### DISCUSSION TITLE
glmm

### DISCUSSION THREAD

**Discussion by AlainZuur** [JenniNiku/gllvm#133] [General]
Hello,
I am writing some text in which I am explaining gllvm at an introductory level. I am trying to do this as didactical as possible. Before diving into models with latent variables, I want to show that you can do ordinary GLMs and GLMMs as well (or so I thought).

My first step is to show that this model:

gl0 <- gllvm(y = SpeciesData, 
                     X = CovData,
                     row.eff = FALSE,
                    formula = ~ Length.std + Sex,
                    num.lv = 0, 
                    family = "negative.binomial",
                    seed = 12345)

is the same as doing multiple NB GLMs on the individual columns of SpeciesData. For example, the output from

nb0 <- glm.nb(Gri ~ Length.std + Sex, data = Fish2)

where Gri is one of the columns of SpeciesData, matches that of the relevant output in gl0. So, that is step 1.

In step 2, I was planning to say that you can also use gllvm to apply GLMMs.

gl1 <- gllvm(y = SpeciesData, 
             X = CovData,
             row.eff = ~(1 | fDayNum),
             formula = ~ Length.std + Sex,
             num.lv = 0, 
             family = "negative.binomial",
             seed = 12345)

It is a data set with multiple observations from the same day, hence the random effect. But this model is not the same as applying GLMMs on the individual species, like this:

M1 <- glmmTMB(Gri ~ Length.std + Sex + (1 | fDayNum),
              family = "nbinom2",
              data = Fish2)

If you apply individual GLMMs, then each species gets its own set of random effects. But gl1 has only one set of random effects for all species.

My question is: How do I fit a gllvm so that I get random effects that differ per species? I tried to include the random effect in the formula, but that was a big boom (crashed R). And I am not sure what  gllvm with   
formula = ~ Length.std + Sex + (1 | DayNum) 
with DayNum as an integer was doing. It did show some estimated parameters that differ per species, but only for 1 species.

The models with latent variables all work fine......

Kind regards,

Alain

**Comment by BertvanderVeen** [comment]
Thanks Alain. Unfortunately gllvm at present does not support additional species-specific random-effects. It should be relatively trivial to implement, but I would place this on the "nice to have" list at present (i.e., other possibilities might have higher priority).

**Reply by AlainZuur** [reply]
Hello Bert. 
Thanks. No problem. It was only for a storyline. I can adjust that.

Alain
